### PR TITLE
feat(report): pdf report #3488

### LINKED ISSUE #3488: Reporting - add PDF report format
**Description**
- Add a new accepted value in the `--report-formats` flag for `pdf`.
- We should find a go native way to convert the HTML report format to PDF.

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
closes #3488

**Proposed Changes**
- PDF report

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/report/pdf.go [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+				})
+			})
+			m.Col(colTwo, func() {
+				m.Text(category, props.Text{
+					Size:        defaultTextSize,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+		})
```
Comment (felipe-avelar) [inline pkg/report/pdf.go:138]: I think we could extract this as function to be something like `createInfoComposition` and replace "Severity", "Platform" and Category row. I would suggest the following snippet to be the func:
```go
func createInfoComposition(m pdf.Maroto, label, info string, rowSize int, config props.Text) {
  m.Row(rowSize, func() {
    m.Col(colTwo, func() {
      m.Text(label, config)
    })
    m.Col(colTwo, func() {
       m.Text(info, config)
    })
  })
}
```
And replace as following:
```go
defaultTextConfig := props.Text{
  Size:        defaultTextSize,
  Align:       consts.Left,
  Extrapolate: false,
}
createInfoComposition(m, "Severity", severity, colFour, props.Text{
  Size:        10,
  Align:       consts.Left,
  Extrapolate: false,
})
createInfoComposition(m, "Platform", platform, colThree, defaultTextConfig)
createInfoComposition(m, "Category", category, colFive, defaultTextConfig)
```

Also, maybe `defaultTextConfig` could be a constant used on multiple locations in this file...

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: pkg/report/pdf.go [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+	colNine         = 9
+	colTen          = 10
+	colFullPage     = 12
+)
+
+var (
+	grayColor = getGrayColor()
+	//go:embed assets/vuln
+	vulnImageBase64 string
+)
```
[the remark this answers is not in the document]
Comment (rogeriopeixotocx) [reply pkg/report/pdf.go:43]: I think it's too much indirection for a small file size

File: pkg/report/pdf.go [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+	}
+}
+
+func getPurpleColor() color.Color {
+	return color.Color{
+		Red:   80,
+		Green: 62,
+		Blue:  158,
+	}
+}
```
[the remark this answers is not in the document]
Comment (rogeriopeixotocx) [reply pkg/report/pdf.go:477]: I think it's too much indirection for a small file size

File: pkg/report/pdf.go [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+	m.Row(rowSmall, func() {
+		m.Col(colTwo, func() {
+			m.Text("PLATFORMS", props.Text{
+				Size:        defaultTextSize,
+				Align:       consts.Left,
+				Extrapolate: false,
+			})
+		})
+		m.Col(colTen, func() {
+			m.Text(getPlatforms(summary.Queries), props.Text{
```
[the remark this answers is not in the document]
Comment (rogeriopeixotocx) [reply pkg/report/pdf.go:292]: good idea

File: pkg/report/pdf.go (hunk 1 above)
Comment (rogeriopeixotocx) [reply pkg/report/pdf.go:138]: good idea

**Review by felipe-avelar — DISMISSED** [review]
LGTM

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**